### PR TITLE
sort: implement --debug

### PR DESCRIPTION
**Pull request by miDeb**
This adds a --debug flag, which, when activated, will draw lines below
the characters that are actually used for comparisons.

This is not a complete implementation of --debug. It should, quoting the man page
for GNU sort: "annotate the part of the line used to sort, and warn
about questionable usage to stderr". Warning about "questionable usage"
is not part of this patch.

This change required some adjustments to be able to get the range that
is actually used for comparisons. Most notably, general numeric comparisons
were rewritten, fixing some bugs along the lines.

Testing is mostly done by adding fixtures for the expected debug output of
existing tests.

### CODE REVIEW

File: src/uu/sort/src/sort.rs [hunk 1]
```diff
@@ -119,6 +121,7 @@ impl Default for GlobalSettings {
     fn default() -> GlobalSettings {
         GlobalSettings {
             mode: SortMode::Default,
+            debug: true,
```
Comment (sylvestre) [inline src/uu/sort/src/sort.rs:124]: I would prefer to have `debug: false,` as default to explicit the default behavior
is it possible to change that?

File: tests/fixtures/sort/multiple_decimals_general.txt [hunk 2]
```diff
@@ -32,4 +32,6 @@ CARAvan
 8.013
 
 000
-
+  NaN
```
Comment (sylvestre) [inline tests/fixtures/sort/multiple_decimals_general.txt:35]: is that expected?

Comment (miDeb) [reply tests/fixtures/sort/multiple_decimals_general.txt:35]: I added a test case for NaN. The expected output is in tests/fixtures/sort/multiple_decimals_general.expected

File: src/uu/sort/src/sort.rs (hunk 1 above)
Comment (miDeb) [reply src/uu/sort/src/sort.rs:124]: Sure, this was for initial testing.

File: tests/fixtures/sort/keys_open_ended.expected.debug [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    ____
+________
+aa bb cc
+    ____
+________
+èè éé èè
+    ____
+________
+👩‍🔬 👩‍🔬 👩‍🔬
+       _______
```
Comment (miDeb) [inline tests/fixtures/sort/keys_open_ended.expected.debug:14]: While this looks off in the browser, it should look correct in a terminal. 👩‍🔬 is actually 👩 +zero_width_joiner+🔬 and is displayed as two characters in my terminal.
This is how it looks in my terminal:
Bildschirmfoto von 2021-04-24 10-57-28

Comment (sylvestre) [reply tests/fixtures/sort/keys_open_ended.expected.debug:14]: the best kind of test :)

**Pull request merged**